### DISCUSSION TITLE
How to upgrade Rehype plugins with new JSDoc based types?

### DISCUSSION THREAD

**Discussion by aurelien-reeves** [rehypejs/rehype#63] [Q&A] [answered]
It seems that new JSDoc based types for rehype plugins - as seen for example with rehype-raw and rehype-sanitize - is not compatible anymore with `unified.Pluggable`

```
Type '(options?: void | Options) => void | Transformer<Root, Root>' is not assignable to type 'Pluggable<any[], Settings>'.
  Type '(options?: void | Options) => void | Transformer<Root, Root>' is not assignable to type 'Plugin<any[], Settings>'.
```

How should we migrate those plugins to their new versions with a typescript codebase?

Refs. https://github.com/cucumber/common/pull/1680
Refs. https://github.com/cucumber/common/pull/1679

**Comment by wooorm** [comment, marked as answer]
Perhaps with a `@ts-expect-error` for now — the actual code should just work but the types depend on a new major of unified (which is already released but isn’t in `react-markdown` yet)

**Reply by aurelien-reeves** [reply]
Thanks, I'll take a look

**Reply by T-Damer** [reply]
Hey, any news about this problem?

**Reply by wooorm** [reply]
Everything is updated. There should not be a problem. If there is a problem, could you explain what it is?

**Reply by T-Damer** [reply]
Well, I'm using [this markdown package](https://github.com/remarkjs/react-markdown) in my React app. When I use custom 
```TS
<ReactMarkdown
            rehypePlugins={[rehypeRaw]}  // <- TS issue here
            children={text}
/>
```
I have same TypeScript issue: 
```
Type '(options?: void | Options | undefined) => void | Transformer<Root, Root>' is not assignable to type 'Pluggable<any[], Settings>'.
```
Adding //@ts-expect-error gives this error on render:
```TS
TypeError: Object(...) is not a function
Module../node_modules/property-information/index.js
.../node_modules/property-information/index.js:17
  14 | export {find} from './lib/find.js'
  15 | export {hastToReact} from './lib/hast-to-react.js'
  16 | export {normalize} from './lib/normalize.js'
> 17 | export var html = merge([xml, xlink, xmlns, aria, htmlBase], 'html')
  18 | export var svg = merge([xml, xlink, xmlns, aria, svgBase], 'svg')
```
Not sure if this problem related to rehype, or to markdown, [this issue](https://github.com/remarkjs/react-markdown/issues/627) led to your discussion

P.S. of course I tried using `as any`, gives same error on render as //@ts-expect-error

**Reply by wooorm** [reply]
Did you try updating `react-markdown`?

**Reply by T-Damer** [reply]
Oh yeah, thank you, hah from `6.0.2` to `6.0.3`, also used `@ts-expect-error` because TS swears :) Also `as any` works too

**Reply by Marujah** [reply]
you can just cast it as as ReactMarkdown.PluggableList:
rehypePlugins={[rehypeRaw] as ReactMarkdown.PluggableList}

**Reply by fieldsco** [reply]
Here's a good option that avoids `any`:

```import { PluggableList } from 'unified'; // already a peer dependency, no need to install```
...
```rehypePlugins={[rehypeRaw, rehypeMathjax] as PluggableList}```

**Comment by Gerald-ux-ux** [comment]
adding `@ts-expect-error` just above the line fixes the solution

**Comment by OpenJarvisAI** [comment]
How to resole it ?

**Reply by wooorm** [reply]
This is 3 years old. It doesn't happen anymore. You are spamming. Post a new question with lots more info.
